src/test_fixture/world.rs

```diff
@@ -144,6 +143,22 @@ impl TestWorld {
             .unwrap()
     }
 
+    /// Creates malicious protocol contexts for 3 helpers
+    ///
+    /// # Panics
+    /// Panics if world has more or less than 3 gateways/participants
+    #[must_use]
+    pub fn malicious_contexts(&self) -> [MaliciousContext<'_>; 3] {
+        let execution = self.executions.fetch_add(1, Ordering::Release);
```

Comment:
my fault, I spelled it wrong inside the similar `contexts` fn. This should be `AcqRel`

Reply:
You are right.  I just copy-pasted.  Will fix.

Reply:
Hang on.  We should be OK here with simply Relaxed.  We don't care that the value is consistent with the surrounding code, just that we get a unique value each time it is invoked.  Relaxed gives us that.

Reply:
My read is that this is OK: https://en.cppreference.com/w/cpp/atomic/memory_order#Relaxed_ordering

That is, the action of incrementing will always occur atomically.  It isn't a load+store where two threads might load the same value; the operation is atomic.  https://stackoverflow.com/a/62613988/1375574 agrees.

Reply:
my reasoning was based on https://doc.rust-lang.org/std/sync/atomic/index.html#portability. Basically not every ARM platform supports atomic CAS and that I believe is reflected in the doc: https://doc.rust-lang.org/std/sync/atomic/struct.AtomicU32.html#method.fetch_add - they explicitly mention ordering for store and add operations.

Neither of those matters in this case I think, so we should leave it as is. Just for actual prod code, I feel like I would still prefer `AcqRel` just to be on the safe side on every platform :) Who knows maybe Android phones will be running IPA soon!

Reply:
I don't see anything in the portability section, other than the note that the implementation might not be performant.  But that is true for any ordering on those platforms.  That is more a question there of availability of primitives, where load+store can cost a lot more if there is contention.

My understanding was that Relaxed has no real benefit on x86, but it does on ARM if there are instructions.